### PR TITLE
Remove duplicate types and reexport from concordium-contracts-common

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Take advantage of types from `concordium-contracts-common`.
Related to https://github.com/Concordium/concordium-base/issues/180.

## Changes

- Update `concordium-base` submodule.
- Remove `concordium-contracts-common` as a direct submodule and instead use it through the `concordium-base` submodule.
- Remove `ReceiveName` and `InitName` in favor of `OwnedReceiveName` and `OwnedContractName`.
- Remove `ContractAddress` and `Address` in favor of the ones in `concordium-contracts-common`.
- Fix examples with the recent changes.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 ## Unreleased changes
 
 - Expose macros for deriving `Serial` and `Deserial` from `concordium-contracts-common`.
+- Address method `is_alias_of` is now `is_alias`.
+- Replaced `ReceiveName` and `InitName` with `OwnedReceiveName` and `OwnedContractName` from `concordium-contracts-common`.
+- Remove `ContractAddress` and `Address` in favor of their equivalents in `concordium-contracts-common`.
+- `AccountAddress::new` is renamed to `account_address_from_registration_id`.
```
Comment (abizjak) [inline CHANGELOG.md:7]: It is also a function now, not a method on a type.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+- `Amount` now have a field `micro_ccd` instead of `microgtu`.
```
Comment (abizjak) [inline CHANGELOG.md:8]:
```suggestion
- `Amount` now has a field `micro_ccd` instead of `microgtu`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
+- The default arithmetic (operator syntax, such as `+`, `-`, `*`) with `Amount` is now unchecked.
+- There are no longer implementations of `From<u64> for Amount` and `From<Amount> for u64` as the behavior of these are not obvious.
```
Comment (abizjak) [inline CHANGELOG.md:10]:
```suggestion
- There are no longer implementations of `From<u64> for Amount` and `From<Amount> for u64` as the behavior of these is not obvious.
```

**Review by abizjak — APPROVED** [review]
Always like it when there are more lines removed than added :)

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,17 @@
 ## Unreleased changes
 
 - Expose macros for deriving `Serial` and `Deserial` from `concordium-contracts-common`.
+- Address method `is_alias_of` is now `is_alias`.
```
Comment (Bargsteen) [inline CHANGELOG.md:4]: Why was this changed? I think the `_of` made more sense, since you give a parameter. `is_alias` sounds more like a `() -> bool` function.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,17 @@
...
+- Replaced `ReceiveName` and `InitName` with `OwnedReceiveName` and `OwnedContractName` from `concordium-contracts-common`.
+- Remove `ContractAddress` and `Address` in favor of their equivalents in `concordium-contracts-common`.
+- `AccountAddress::new` is replaced by a function called `account_address_from_registration_id`.
+- `Amount` now has a field `micro_ccd` instead of `microgtu`.
+- The default arithmetic (operator syntax, such as `+`, `-`, `*`) with `Amount` is now unchecked.
```
Comment (Bargsteen) [inline CHANGELOG.md:9]: What is the reasoning behind this change?

**Review by Bargsteen — APPROVED** [review]
LGTM 👍

File: CHANGELOG.md (hunk 4 above)
Comment (limemloh) [reply CHANGELOG.md:4]: I agree, but changing it the other way around would mean breaking changes in `concordium-contracts-common`.

File: CHANGELOG.md (hunk 5 above)
Comment (limemloh) [reply CHANGELOG.md:9]: The reason being that the implementation of these operators are already present in `concordium-contracts-common` but with the unchecked.
But, I also think, the default arithmetic should not be checked.

File: CHANGELOG.md (hunk 4 above)
Comment (Bargsteen) [reply CHANGELOG.md:4]: Ah, okay. Np then ;)

**Review by abizjak — APPROVED** [review]
